server/message_queue_service.py

```diff
@@ -23,6 +23,8 @@ def __init__(self) -> None:
         self._channel = None
         self._exchanges = {}
         self._exchange_types = {}
+        self._is_ready = False
+        self._initialization_lock = asyncio.Lock()
```

Comment:
Btw I wrote a decorator for this here https://github.com/FAForever/server/pull/629/commits/cda403335f72b7ca428ef6cb553c8bbc1c963193. Maybe I should extract it to a separate PR?

It lets you do:
```python3
class SomeClass:

    @synchronizedmethod
    async def some_func(self, args):
        ... 
```
to wrap a method with one lock per instance, or 
 
```python3
@synchronized
async def some_func(args):
    ...
```
to lock a function with a global lock (works for methods too, but you will only be able to call the method on one instance at a time).

Reply:
Oh I guess you are using the lock in more than one function so maybe it's not useful here.

Reply:
I guess one could add something like `@synchronized_group(lock_name)` managing a global dictionary of locks that can be shared between functions/methods, if you want.

Reply:
I worked on it a bit more, and I came up with that same solution, except I just made it an optional parameter for the existing decorator. So it would be:

```python3
class Foo:
    def __init__(self):
        self._the_lock = asyncio.Lock()

    @synchronizedmethod("_the_lock")
    async def bar(self):
        ...

    @synchronizedmethod(lock="_the_lock")
    async def baz(self):
        ...

```